src/nats/mod.rs

```diff
@@ -0,0 +1,222 @@
+}
+
+impl Nats {
+    const DEFAULT_USER: &'static str = "";
+    const DEFAULT_PASS: &'static str = "";
+    const DEFAULT_VERSION_TAG: &'static str = "2.10.14";
+
+    /// Create a new instance of a Nats image.
+    #[must_use]
+    pub fn new() -> Self {
+        Self {
+            version: Cow::Borrowed(Self::DEFAULT_VERSION_TAG),
+            user: Some(Cow::Borrowed(Self::DEFAULT_USER)),
+            pass: Some(Cow::Borrowed(Self::DEFAULT_PASS)),
+        }
+    }
+
+    /// Set the username to use.
+    #[must_use]
+    pub fn with_user(mut self, user: impl Into<Value>) -> Self {
+        self.user = Some(user.into());
+        self
+    }
+
+    /// Set the password to use.
+    #[must_use]
+    pub fn with_password(mut self, pass: impl Into<Value>) -> Self {
+        self.pass = Some(pass.into());
+        self
+    }
+
+    /// Do not use any authentication on the testcontainer.
+    ///
+    /// Setting this will override any prior usages of [`Self::with_user`] and
+    /// [`Self::with_password`].
+    pub fn without_authentication(mut self) -> Self {
+        self.user = None;
+        self.pass = None;
+        self
+    }
+}
+
+type Value = Cow<'static, str>;
+
+impl Default for Nats {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+/// The actual Nats testcontainers image type which is returned by `container.image()`
+pub struct NatsImage {
+    version: String,
+    auth: Option<(String, String)>,
+}
+
+impl NatsImage {
+    /// Return the version of the Nats image.
+    #[must_use]
+    pub fn version(&self) -> &str {
+        &self.version
+    }
+
+    /// Return the user/password authentication tuple of the Nats server.
+    /// If no authentication is set, `None` is returned.
+    #[must_use]
+    pub fn auth(&self) -> Option<(&str, &str)> {
+        self.auth
+            .as_ref()
+            .map(|(user, pass)| (user.as_str(), pass.as_str()))
+    }
+
+    /// Return the user of the Nats server.
+    /// If no authentication is set, `None` is returned.
+    #[must_use]
+    pub fn user(&self) -> Option<&str> {
+        self.auth().map(|(user, _)| user)
+    }
+
+    /// Return the password of the Nats server.
+    /// If no authentication is set, `None` is returned.
+    #[must_use]
+    pub fn password(&self) -> Option<&str> {
+        self.auth().map(|(_, pass)| pass)
+    }
+}
+
+impl Image for NatsImage {
+    type Args = ();
+
+    fn name(&self) -> String {
+        "nats".to_owned()
+    }
+
+    fn tag(&self) -> String {
+        self.version.clone()
+    }
+
+    fn ready_conditions(&self) -> Vec<WaitFor> {
+        vec![
+            WaitFor::message_on_stderr("Listening for client connections on 0.0.0.0:4222"),
+            WaitFor::message_on_stderr("Server is ready"),
+        ]
+    }
+}
+
+impl Nats {
+    pub fn build(self) -> NatsImage {
+        let auth = self
+            .user
+            .and_then(|user| self.pass.map(|pass| (user.into_owned(), pass.into_owned())));
```

Comment:
Btw, there is a shorter syntax for such operation: [zip method](https://doc.rust-lang.org/core/option/enum.Option.html#method.zip)

Reply:
This method has been removed: [bca942f](https://github.com/testcontainers/testcontainers-rs-modules-community/pull/123/commits/bca942f0af1f4f393a818ae132cc83c4b825415e)